docs/topics/properties.md

```diff
@@ -67,156 +131,353 @@ fun main() {
+    val temp = Temperature()
+    temp.celsius = 25.0
+    println("${temp.celsius}°C = ${temp.fahrenheit}°F") // 25.0°C = 77.0°F
+
+    temp.fahrenheit = 212.0
+    println("${temp.celsius}°C = ${temp.fahrenheit}°F") // 100.0°C = 212.0°F
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-backing-property-multiple-properties"}
+
+In this example, the backing property `_celsius` is accessed by both the `celsius` and `fahrenheit` properties. So there 
+is one source of truth but two public views.
 
 ## Compile-time constants
 
-If the value of a read-only property is known at compile time, mark it as a _compile time constant_ using the `const` modifier.
-Such a property needs to fulfill the following requirements:
+If the value of a read-only property is known at compile time, mark it as a _compile-time constant_ using the `const` modifier.
+Compile-time constants are inlined at compile time, so every reference is replaced with its actual value. They are accessed
+more efficiently as no getter is called and no memory is allocated for the property.
+
+```kotlin
+// File: AppConfig.kt
+package com.example
+
+// Compile-time constant
+const val MAX_LOGIN_ATTEMPTS = 3
+```
+
+Compile-time constants must satisfy the following requirements:
+
+* They must be either a top-level property, or a member of an [`object` declaration](object-declarations.md#object-declarations-overview) or a _[companion object](object-declarations.md#companion-objects)_.
+* They must be initialized with a value of type `String` or a [primitive type](basic-types.md).
+* They can't have a custom getter.
 
-* It must be a top-level property, or a member of an [`object` declaration](object-declarations.md#object-declarations-overview) or a _[companion object](object-declarations.md#companion-objects)_.
-* It must be initialized with a value of type `String` or a primitive type
-* It cannot be a custom getter
+> A top-level property is a property declared directly in a `.kt` file, not inside a class, interface, or object.
+> Think of it as a global variable or constant that belongs to the package.
+> 
+{style="tip"}
 
-The compiler will inline usages of the constant, replacing the reference to the constant with its actual value. However, the field will not be removed and therefore can be interacted with using [reflection](reflection.md).
+Compile-time constants still have a backing field, so you can interact with them using [reflection](reflection.md).
 
-Such properties can also be used in annotations:
+These properties have the added benefit that they can also be used in annotations:
 
 ```kotlin
 const val SUBSYSTEM_DEPRECATED: String = "This subsystem is deprecated"
 
-@Deprecated(SUBSYSTEM_DEPRECATED) fun foo() { ... }
+@Deprecated(SUBSYSTEM_DEPRECATED) fun processLegacyOrders() { ... }
 ```
 
 ## Late-initialized properties and variables
 
 Normally, properties declared as having a non-nullable type must be initialized in the constructor.
```

Comment:
Properties with nullable type must be initialized in the constructor as well. In the context of initialization, nullability doesn't matter.